src/main/java/org/openrewrite/java/testing/cucumber/CucumberJava8StepDefinitionToCucumberJava.java

```diff
@@ -0,0 +1,158 @@
+            List<Expression> arguments = methodInvocation.getArguments();
+            if (arguments.size() < 2) {
+                return methodInvocation;
+            }
+
+            // Annotations require a String literal
+            Expression stringExpression = arguments.get(0);
+            if (!(stringExpression instanceof J.Literal)) {
+                return methodInvocation
+                        .withMarkers(methodInvocation.getMarkers().searchResult("TODO Migrate manually"));
+            }
+            J.Literal literal = (J.Literal) stringExpression;
+
+            // Extract step definition body, when applicable
+            Expression possibleStepDefinitionBody = arguments.get(1); // Always available after a first String argument
+            if (!(possibleStepDefinitionBody instanceof J.Lambda)
+                    || !TypeUtils.isAssignableTo(IO_CUCUMBER_JAVA8_STEP_DEFINITION_BODY,
+                            possibleStepDefinitionBody.getType())) {
+                return methodInvocation
+                        .withMarkers(methodInvocation.getMarkers().searchResult("TODO Migrate manually"));
+            }
+            J.Lambda lambda = (J.Lambda) possibleStepDefinitionBody;
+
+            StepDefinitionArguments stepArguments = new StepDefinitionArguments(
+                    methodInvocation.getSimpleName(), literal, lambda);
+
+            // Determine step definitions class name
+            J.ClassDeclaration parentClass = getCursor()
+                    .dropParentUntil(J.ClassDeclaration.class::isInstance)
+                    .getValue();
+            String replacementImport = String.format("%s.%s",
+                    methodInvocation.getMethodType().getDeclaringType().getFullyQualifiedName()
+                            .replace("java8", "java").toLowerCase(),
+                    methodInvocation.getSimpleName());
+            doAfterVisit(new CucumberJava8ClassVisitor(
+                    parentClass.getType(),
+                    replacementImport,
+                    stepArguments.template(),
+                    stepArguments.parameters()));
+
+            // Remove original method invocation; it's replaced in the above visitor
+            return null;
+        }
+    }
+
+}
+
+@Value
+class StepDefinitionArguments {
+
+    String annotationName;
+    J.Literal cucumberExpression;
+    J.Lambda lambda;
+
+    String template() {
+        return "@#{}(#{any()})\npublic void #{}(#{}) throws Exception {\n\t#{any()}\n}";
+    }
+
+    private String formatMethodName() {
+        return ((String) cucumberExpression.getValue())
+                .replaceAll("\\s+", "_")
+                .replaceAll("[^A-Za-z0-9_]", "")
+                .toLowerCase();
+    }
+
+    private String formatMethodArguments() {
+        // TODO Type loss here, but my attempts to pass these as J failed: __P__.<java.lang.Object>/*__p0__*/p <error>()
```

Comment:
Imports are not an issue in practice, as templates will use the original imports,
but any comments attached to lambda arguments (unlikely) are lost.